core/src/main/java/org/infinispan/configuration/cache/TransactionConfigurationBuilder.java

```diff
@@ -319,6 +324,12 @@ public void validate() {
       if (!attributes.attribute(NOTIFICATIONS).get()) {
          log.transactionNotificationsDisabled();
       }
+      if (attributes.attribute(TRANSACTION_MODE).get() == TransactionMode.TRANSACTIONAL && !cacheMode.isSynchronous()) {
+         //change to sync or throw exception?
+         CacheMode syncCacheMod = cacheMode.toSync();
+         log.unsupportedAsyncCacheMode(cacheMode, syncCacheMod);
```

Comment:
+1 for throwing an exception. As a user I would prefer to fail-fast and explicitly.